### PR TITLE
Issue and PR requirements (templates)

### PR DESCRIPTION
**Pull request by adrianhall**
* Added templates for issues related to design guidelines
* Added CODEOWNERS for all directories with the appropriate reviewers
* CODEOWNERS also provides review requirements

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Engineering systems documents
+/docs/engineering-system     @maggiepint,@Kurtzeborn,@weshaggard,@mikeharder
+
+# General Doc Guidelines
+/docs/design                 @bterlson,@JeffreyRichter,@JonathanGiles,@KrzysztofCwalina,@johanste,@gornishanov,@adrianhall,@petermarcu,@loarabia
+
+# Per-language Guidelines
+/docs/design/dotnet          @KrzysztofCwalina,@JeffreyRichter,@davidfowl,@AlexGhiondea,@adrianhall,@petermarcu,@loarabia
+/docs/design/java            @JonathanGiles,@selvasingh,@brunoborges,@mayurid,@adrianhall,@petermarcu,@loarabia
+/docs/design/python          @johanste,@annatisch,@mayurid,@adrianhall,@petermarcu,@loarabia
```
Comment (johanste) [inline .github/CODEOWNERS]: Please add @bryevdv for Python.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#####
+##### TEAM:
+#####
+#####   Architecture:
+#####       Brian Terlson               @bterlson           TypeScript / JavaScript
+#####       Jeffrey Richter             @JeffreyRichter     At large
+#####       Jonathan Giles              @JonathanGiles      JavaScript
```
Comment (JonathanGiles) [inline .github/CODEOWNERS]:
```suggestion
#####       Jonathan Giles              @JonathanGiles      Java
```

File: .github/ISSUE_TEMPLATE/adp_review.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+* Link to documentation describing the service:
+* Contact email (if service team, provide PM and Dev Lead):
+* Expected code complete date:
+* Expected release date:
+
+## About this client library
+
+* Name of the client library:
+* Target language for this review:
+* Other languages that are to be targetted:
```
Comment (JonathanGiles) [inline .github/ISSUE_TEMPLATE/adp_review.md]:
```suggestion
* Other languages that are to be targeted:
```

File: .github/ISSUE_TEMPLATE/adp_review.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+assignees: adrianhall
+---
+
+Thank you for starting the process for approval of the client library for your Azure service.  Thorough review of your client library ensures that your APIs are consistent with the guidelines and the consumers of your client library have a consistently good experience when using Azure. 
+
+** Before submitting, ensure you adjust the title of the issue appropriately **
+
+## The Basics
+
+* Service team is responsible for the client library:
```
Comment (JonathanGiles) [inline .github/ISSUE_TEMPLATE/adp_review.md]:
```suggestion
* Service team responsible for the client library:
```

**Pull request merged**